libs/design/src/molecules/accordion/accordion-item/accordion-item.component.ts

```diff
@@ -1,15 +1,30 @@
-import { Component, Input, OnInit, ViewEncapsulation } from '@angular/core';
+import { Component, Input, OnInit, ViewEncapsulation, OnChanges } from '@angular/core';
+
+import { daffAccordionAnimations } from '../animation/accordion-animation';
+import { getAnimationState } from '../animation/accordion-animation-state';
+
 
 @Component({
-  selector: 'accordion-item',
+  selector: 'daff-accordion-item',
+  host: {'class': 'daff-accordion-item'},
   templateUrl: './accordion-item.component.html',
   styleUrls: ['./accordion-item.component.scss'],
-  encapsulation: ViewEncapsulation.None
+  encapsulation: ViewEncapsulation.None,
+  animations: [
+    daffAccordionAnimations.openAccordion
+  ]
 })
-export class AccordionItemComponent implements OnInit {
+export class DaffAccordionItemComponent implements OnInit {
   @Input() initiallyActive: boolean;
+  _open: boolean = false;
+  get open(): boolean {
+    return this._open;
+  };
 
-  open: boolean;
+  set open(value: boolean) {
+    this._open = value;
+    this._animationState = getAnimationState(this._open);
+  }
```

Comment:
I think there should be a cleaner way to handle the open variable. In other design components we needed the getter and setter, because state was coming from the user, and we needed to hold that in a private variable. Unlike a lot of the other design components, this one holds the open state itself, which I think makes sense in this case.  So it shouldn't need to have an `_open` and a getter and setter for open.

Reply:
I'm not 100% sure on this though, and @damienwebdev should probably take a look too

Reply:
I just tried this and it works:
```
@Input() initiallyActive: boolean;
  _open: boolean;
  _animationState: string;
  
  ngOnInit() {
    this._open = this.initiallyActive;
    this._animationState = getAnimationState(this._open);
  }

  toggleActive() {
    this._open = !this._open;  
    this._animationState = getAnimationState(this._open);
  }
```

Reply:
@lderrickable I agree. I'm waffling back and forth a bit on this. I like the concept of ALL our components being "functional" (they don't manage their own state), but it's hard to weigh that against "up-front" dev-user experience.

Tracking state in the component here is acceptable for now, but I can forsee a future where we have an `@Input() open`.

I'll agree with your changes for now, but I'm not strongly against changing this in the future if there is a use-case for it.

Reply:
Resolved in 21ccdc7